### ISSUE TITLE
some gce/gke-specific e2e tests are being unintentionally skipped

### ISSUE THREAD

**Issue by ixdy** [GoogleCloudPlatform/kubernetes#7008]
I noticed while running e2e tests locally that some tests were apparently being incorrectly skipped:
```
Setting up for KUBERNETES_PROVIDER="gce".
Project: [project]
Zone: [zone]
STEP: Skipping guestbook, uses createExternalLoadBalancer, a (gce|gke) feature
STEP: Skipping Shell test, which is only supported for provider gce and gke (not )
I0417 15:33:42.606949   19035 driver.go:61] ***** provider is 'gce'
Running Suite: Kubernetes e2e Suite run 1 of 1
==============================================
Random Seed: 1429310022 - Will randomize all specs
Will run 34 of 39 specs
...
```

I checked, and this is also true for our Jenkins runs.

Currently it looks like test/e2e/shell.go and test/e2e/kubectl.go have this bug, though only the former actually skips the test, I think.

The issue appears to be that the code inside the top-level `Describe()` in these files is being run before the testContext struct is filled out (and before `ginkgo.RunSpecsWithDefaultAndCustomReporters` is even called), and so testContext.Provider is empty.

I'm not immediately sure what the proper fix is; I'm also worried that there might be other things broken by the fact that these blocks are called before initialization is complete.

We also probably investigate if there is a better way to mark these tests as skipped; right now, there's no indication in the test report (JUnit.xml, Jenkins test report, etc) that these tests were skipped.

**Comment by roberthbailey** [comment]
I merged #6817 and #6875 which tried to disable these tests for non-{gke,gce} but apparently didn't follow-up and verify that they were still running on our platforms. /cc @timothysc

**Comment by roberthbailey** [comment]
Seems easy enough to fix. I'll have a PR out to fix this before Monday.
